File: src/discord/commands/public/fab.ts
import { createCommand } from "#base";
import { fabComponent } from "#menus";
import { ApplicationCommandType, InteractionContextType } from "discord.js";

createCommand({
    name: "fab",
    description: "Products on fab marketplace",
    contexts: [ InteractionContextType.Guild ],
    type: ApplicationCommandType.ChatInput,
    async run(interaction) {
        const component = fabComponent();

        interaction.reply({ flags: [ "IsComponentsV2", "Ephemeral" ], components: [ component ] });
    }
})

File: src/discord/commands/public/fab.rs
use crate::discord::*;
use crate::menus::*;
use async_trait::async_trait;
use serenity::all::{
    CommandInteraction, CommandType, Context, CreateCommand, InteractionContext, MessageFlags,
};

pub struct Fab;

#[async_trait]
impl SlashCommandHandler for Fab {
    fn command(&self) -> CreateCommand<'static> {
        CreateCommand::new("fab")
            .description("Products on fab marketplace")
            .kind(CommandType::ChatInput)
            .add_context(InteractionContext::Guild)
    }

    async fn run(&self, app: &App, ctx: &Context, interaction: &CommandInteraction) {
        let component = fab_component();
        let payload = ReplyPayload {
            components: Some(vec![component]),
            ..ReplyPayload::default()
        };
        reply(ctx, interaction, MessageFlags::IS_COMPONENTS_V2, &payload).await;
    }
}
